server/conn_test.go

```diff
@@ -132,6 +132,37 @@ func (ts ConnTestSuite) TestInitialHandshake(c *C) {
 	c.Assert(outBuffer.Bytes()[4:], DeepEquals, expected.Bytes())
 }
 
+func (ts ConnTestSuite) TestInitialHandshakeWithCollation(c *C) {
```

Comment:
The test doesn't make much sense as the `writeInitialHandshake` always send `mysql.DefaultCollationID`.
We should test the session system variable has changed according to the collation in `handshakeResponse`.

Reply:
ok, later